src/utils/narrow.js

```diff
@@ -148,11 +148,12 @@ export const ALL_PRIVATE_NARROW: Narrow = specialNarrow('private');
 
 export const ALL_PRIVATE_NARROW_STR: string = keyFromNarrow(ALL_PRIVATE_NARROW);
 
-export const streamNarrow = (stream: string): Narrow =>
-  Object.freeze({ type: 'stream', streamName: stream });
+export const streamNarrow = (streamName: string, streamId?: number): Narrow =>
+  Object.freeze({ type: 'stream', streamName });
 
-export const topicNarrow = (stream: string, topic: string): Narrow =>
-  Object.freeze({ type: 'topic', streamName: stream, topic });
+export const topicNarrow = (streamName: string, ...args: [string] | [number, string]): Narrow =>
+  // $FlowFixMe[incompatible-cast]
+  Object.freeze({ type: 'topic', streamName, topic: (args[args.length - 1]: string) });
```

Comment:
Yeah, exactly.

If this were going to last longer -- if this were going to be a longer transition -- then I'd want to take a strategy that would be somewhat more overhead but keep the intermediate states more robust. In particular I'd probably have two different constructors for the two different signatures as long as they coexisted -- so like:
1. Rename `topicNarrow` to like `topicNarrowOld`, or `topicNarrowNoId`.
2. Add new `topicNarrow` with the new signature. (Or even: leave the old name nonexistent, so other branches rebased across the change would definitely get a useful error.)
3. Go converting things from using the old `topicNarrowNoId` to using the new `topicNarrowWithId`. This might go on for a while.
4. Eventually, delete the old thing.
5. Eventually, rename the new thing to the simple name (if it didn't get it at step 2.)

That's the kind of approach that can scale to a larger codebase with lots of different people working in it, or a codebase that isn't all versioned together in one repo. And I've used it for some changes even in zulip-mobile, where the conversions in step 3 are tangly. But it's more work and churn and I was glad this change didn't end up needing it.